### ISSUE TITLE
TypeError: unicode strings are not supported, please encode to bytes: 'mw 80500000 0\n'

### ISSUE THREAD

**Issue by ncjsvr** [HorstBaerbel/ubootwrite#3]
Hi,

I am getting the following error when trying to use ubootwrite.py on a Meraki MR33.

C:\Users\Nathan\Desktop\mr33>python ubootwrite.py --serial=COM5 --write=mr33-uboot.bin
Traceback (most recent call last):
  File "ubootwrite.py", line 173, in <module>
    main()
  File "ubootwrite.py", line 169, in main
    memwrite(ser, options.write, int(options.size, 0), int(options.addr, 0), options.verbose, debug)
  File "ubootwrite.py", line 71, in memwrite
    prompt = getprompt(ser, start_addr, verbose)
  File "ubootwrite.py", line 23, in getprompt
    ser.write("mw {0:08x} 0".format(addr) + LINE_FEED)
  File "C:\Program Files (x86)\Python38-32\lib\site-packages\serial\serialwin32.py", line 308, in write
    data = to_bytes(data)
  File "C:\Program Files (x86)\Python38-32\lib\site-packages\serial\serialutil.py", line 63, in to_bytes
    raise TypeError('unicode strings are not supported, please encode to bytes: {!r}'.format(seq))
TypeError: unicode strings are not supported, please encode to bytes: 'mw 80500000 0\n'

**Comment by HorstBaerbel** [comment]
Not sure what's wrong. Did you clone the project or did you save the file from the web interface? You could try setting the proper line feed for Windows in [Line 17](https://github.com/HorstBaerbel/ubootwrite/blob/819676e036113bda44548256f1fb09e892813405/ubootwrite.py#L17):
`LINE_FEED = os.linesep`

**Comment by twopercent** [comment]
I got a very similar error with Python 3.8.2 and pyserial 3.4
I made a little progress adding `str.encode()` to the ser.write commands as shown in this example
https://stackoverflow.com/questions/35642855/python3-pyserial-typeerror-unicode-strings-are-not-supported-please-encode-to
`ser.write(str.encode("mw {0:08x} 0".format(addr) + LINE_FEED))`

Next I got errors for concatenating strings and bytes in some print statements.
I resolved this by adding a `b` in front of the strings, e.g.
`print(b"Prompt is '" + buf[2:] + b"'")`

Now it runs until:
Found an error, so aborting

So I am not sure if my device is bricked, my changes have broken something, or there are other updates needed to the script.

**Comment by HorstBaerbel** [comment]
I'm sorry, but I can't test ANY of this. I'd suggest you try out brntool or maybe the Python3 fixes [here](https://github.com/rvalles/brntool/pull/9/commits/07dbce822e5c69412e28504e638c4e9f23bf4767) and [here](https://github.com/rvalles/brntool/pull/5/commits/d50da30689b955e2dee23b256b34522004d6269c?diff=split&w=1). Pull requests are very welcome.

**Comment by voyager529** [comment]
Hello there!

I'm having the same issue as Nathz...and sadly, this is the only place there seems to be a thread directly related to the error. 

Same objective - flash OpenWRT on an MR33. I have the serial connection just fine; it's running on Mobaxterm and I can see the output (I'm on Windows). 

I've tried Python 2.7.18, 3.6.8, 3.8.5, and pySerial 3.0 and 3.4 in basically every combination. The MR33 is indeed running the proper firmware for flashing.

I tried the line feed change on Line17 described above, to no avail.

Other times I've gotten beyond that, to get this instead:
```
Traceback (most recent call last):
  File "ubootwrite.py", line 175, in <module>
    main()
  File "ubootwrite.py", line 159, in main
    ser = serial.Serial(options.serial, 115200, timeout=0.1)
AttributeError: 'module' object has no attribute 'Serial'
```
I looked at twopercent's option, but I'm just not certain where to even try that; my programming expertise basically begins and ends with the occasional batch script. Consequently, utilizing brntool wasn't much of an option; while your script is mentioned in the MR33 flashing guide on the OpenWRT forum and the syntax is listed, brntool lacks a similar "copy/paste this onto a command line" example to draw from.

If you do have any other wisdom that can be provided, I would appreciate it...and if I do manage to flash this access point, I'll be more than happy to report back my procedures.
